envconfig.go

```diff
@@ -217,6 +218,7 @@ type options struct {
 	Overwrite bool
 	Required  bool
 	Prefix    string
+	NoInit    bool
```

Comment:
I've thought about it, and I think we should have the tag also be "noinit". We have clear documentation that says "even with noinit, the struct is still initialized if a child value has an env".